### ISSUE TITLE
Detailed information about training data and test data

### ISSUE THREAD

**Issue by baizeweiyu** [lin-tan/CoCoNut-Artifact#7]
Hello, thank you for sharing the code and data. Because I found that I need to prepare the test set by myself during the test, I would like to ask, in the training and test data, there are 4 .txt files, rem, add, meta and context. Does rem refer to a bug? What is add? Can it refer to fix? The meta file includes a data and path. What does that data represent and what role does the meta file play in training? Thank you for your answer! Many thanks!

**Comment by thibolu** [comment]
Hello,
The shared data was extracted from bug fixing commits:

rem.txt contains the buggy lines 
add.txt contains the fixed line.

the three files add, rem, meta have a line-to-line mapping. For example, the fix for the bug on line 10 of rem.txt is  on line 10 of add.txt.

The meta is not used in training. It's just a help to identify buggy and fixed lines during debugging and data cleaning. It can be used (although it's a bit tedious) to identify the original commit using the commit sha.

The README attached with the release contains additional information and examples.

Let me know if you have further questions.

**Comment by baizeweiyu** [comment]
Hello: I am sorry, I downloaded the ReadMe file after I downloaded the training data, thank you for your answer!I have no problem with the time being!Thank you so much!(๑`▽´๑)۶



---Original---
From: ***@***.***&gt;
Date: Wed, Dec 29, 2021 21:07 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lin-tan/CoCoNut-Artifact] Detailed information about training data and test data (Issue #7)




 
Hello,
 The shared data was extracted from bug fixing commits:
 
rem.txt contains the buggy lines
 add.txt contains the fixed line.
 
the three files add, rem, meta have a line-to-line mapping. For example, the fix for the bug on line 10 of rem.txt is  on line 10 of add.txt.
 
The meta is not used in training. It's just a help to identify buggy and fixed lines during debugging and data cleaning. It can be used (although it's a bit tedious) to identify the original commit using the commit sha.
 
The README attached with the release contains additional information and examples.
 
Let me know if you have further questions.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nashid** [comment]
@baizeweiyu you were able to train the model with the dataset? What steps did you take?

**Comment by baizeweiyu** [comment]
After decompression, modify the directory in the code, then run the script, including preprocessing data training, and then test.However, it is currently in a state of training, and the specific data has not been analyzed.


---Original---
From: ***@***.***&gt;
Date: Thu, Jan 13, 2022 17:34 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lin-tan/CoCoNut-Artifact] Detailed information about trainingdata and test data (Issue #7)




 
@baizeweiyu you were able to train the model with the dataset? What steps did you take?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nashid** [comment]
@baizeweiyu I presume you are referring to the following steps?

- unzip data like for `python.tar.lrz`
- the directory in the code, then run the script for including preprocessing `generate_data.py`
- Then I should run `train.py`
- And then finally test script. There are multiple scripts - which one?

Also for running script even after setting the correct path, I presume you have to make many changes as the code does not work as is. Is this correct?

**Comment by baizeweiyu** [comment]
The decompression step is correct, then run tokenize.sh, train_all.sh and test_all.sh.Where you need to confirm that the directory is correct, it is probably the case.




---Original---
From: ***@***.***&gt;
Date: Fri, Jan 14, 2022 06:19 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lin-tan/CoCoNut-Artifact] Detailed information about trainingdata and test data (Issue #7)




 
@baizeweiyu I presume you are referring to the following steps?
  
unzip data like for python.tar.lrz 

the directory in the code, then run the script for including preprocessing generate_data.py 

Then I should run train.py 

And then finally test script. There are multiple scripts - which one? 
 
Also for running script even after setting the correct path, I presume you have to make many changes as the code does not work as is. Is this correct?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nashid** [comment]
@baizeweiyu I was able to run the `train` but not `test` script. Do you have to do anything specific to run the `test` script?

**Comment by nashid** [comment]
@baizeweiyu not sure how you were able to run the test script! This artefact is not complete and unless I am missing something don't see how to run the test dataset.

Also, the test dataset is not shared along with the artefact. Only partial is shared [here](https://github.com/lin-tan/CoCoNut-Artifact/releases/tag/training_data_1.0.0).

As an example, `rem.txt`, `add.txt`, `meta.txt`, `context.txt` for the `manybugs` is not shared here:

```
    rem_path = './data/test//origin/rem.txt'
    add_path = './data/test/manybugs/origin/add.txt'
    meta_path = './data/test/manybugs/origin/meta.txt'
    context_path = './data/test/manybugs/origin/context.txt'
```
